### DISCUSSION TITLE
Setting up foundry for frontend development (run foundry persistently listening on a certain port)

### DISCUSSION THREAD

**Discussion by 0xTimepunk** [foundry-rs/foundry#4465] [Q&A] [answered]
Hi,
Is it possible to deploy a set of contracts against a RPC fork and keep it running just like it was possible with ganache? (Listening on a certain port)

How?

Thanks

**Comment by burnto** [comment, marked as answer]
Yes, for sure, with `anvil`. Here's what I do:

```
anvil --fork-url <YOUR_RPC_ENDPOINT> --fork-block-number <YOUR_BLOCK>  --chain-id 1337
```

This starts up `anvil` forked against another chain.

Then call scripts against it. You could use hardhat or ganache to deploy, or here's how you'd do it with foundry:

```
forge script script/00_ExampleDeploy.sol \
  --sender <FIRST_ADDRESS_IN_ANVIL_OUTPUT>  \
  --private-key <FIRST_PRIVATE_KEY_IN_ANVIL_OUTPUT> \
  --rpc-url http://127.0.0.1:8545/ \
  --broadcast
```

In your script you could also use CREATE2 (via `new MyContract{salt: mySalt}()`) to ensure the same address across network starts/stops.
